docs/proposals/message-proto-design.md

```diff
@@ -0,0 +1,296 @@
+---
+title: change model message to  protobuf message 
+authors:
+    - "@kadisi"
+    - "@rohitsardesai83"
+approvers:
+  - "@CindyXing"
+  - "@sids-b"
+  - "@qizha"
+  - "kevin-wangzefeng"
+creation-date: 2019-07-11
+last-updated: 2019-07-24
+status: alpha
+---
+
+# change model message to protobuf message 
+
+* [change model message to protobuf message](#change-model-message-to-protobuf-message)
+   * [Abstract](#abstract)
+   * [Need to do](#need-to-do)
+   * [How to do](#how-to-do)
+   * [Change the model message to protobuf message  directly](#change-the-model-message-to-protobuf-message--directly)
+   * [Solve the filed index problem of Resource](#solve-the-filed-index-problem-of-resource)
+   * [Consider compatibility](#consider-compatibility)
+* [How to use in KubeEdge project](#how-to-use-in-KubeEdge-project)
+* [Which projects need to be modified](#which-projects-need-to-be-modified)
+
+## Abstract 
+
+currently, In beehive framework, model message is defined and serialized using json format. the performance is slower than protobuf.
+
+and for parsing the message resource, each field index is hardcoded. such as (https://github.com/kubeedge/beehive/blob/master/pkg/common/util/parse_resource.go)
```

Comment:
And for parsing the message resource, each field index is hardcoded, such as (https://github.com/kubeedge/beehive/blob/master/pkg/common/util/parse_resource.go).

Reply:
ok